build.gradle

```diff
@@ -206,6 +217,19 @@ subprojects {
       } 
     }
   }
+
+  publishing {
+     publications {
+        mavenJava(MavenPublication) {
+	   groupId "org.apache.edgent"
+           artifactId "$project.group".replace("edgent.", "") + "." + artifact
+	   version build_version
+           artifact sourceJar
+
+	   from components.java
+        }
+     }
+  }
```

Comment:
The other day, I was playing with this, came to appreciate your comment about the unique groups, and I tried doing essentially the same thing :-)
I thought specifiers like  `org.apache.edgent:edgent.api.topology:0.4.0` would be better
I had migrated setting the project's version to allprojects ( `project.version = build_version`) so I didn't need the version spec as it defaults to the project version.
And I added `build_group: org.apache.edgent` to `gradle.properties` just to keep those identification strings in one place.
That netted out to:
```
  publishing {
     publications {
        mavenJava(MavenPublication) {
           from components.java
           // specify dependencies like: org.apache.edgent:edgent.api.topology:0.4.0
           groupId = build_group
           artifactId = "${project.group}.${project.name}" 
           artifact sourceJar
        }
     }
  }  
```
Does that naming scheme / impl seem good to you?